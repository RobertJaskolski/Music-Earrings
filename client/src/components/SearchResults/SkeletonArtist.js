import React from "react";
import Skeleton from "@material-ui/lab/Skeleton";
import styled from "styled-components";

const StyledSkeletonIMG = styled(Skeleton)`
  position: absolute;
  background-color: #1db954;
`;
const StyledSkeletonText = styled(Skeleton)`
  height: 35px;
  background-color: #1db954;
  margin-left: 55px;
`;

const Chip = styled.div`
  flex-basis: 95%;
  border-radius: 5px;
  margin-bottom: 25px;
  opacity: 0.8;
`;

export default function SkeletonArtist() {
  return (
    <Chip>
      <StyledSkeletonIMG variant='rect' width={55} height={55} />
      <StyledSkeletonText variant='rect' />
    </Chip>
  );
}
