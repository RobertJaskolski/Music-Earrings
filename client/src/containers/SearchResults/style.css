.hidden {
  transform: scaleY(0);
}
