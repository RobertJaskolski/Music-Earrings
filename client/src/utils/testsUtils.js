import checkPropTypes from "check-prop-types";
export { default as testStore } from "../store";
export const findByDataTest = (component, attr) => {
  const wrapper = component.find(`[data-test='${attr}']`);
  return wrapper;
};

export const checkProps = (component, expectedProps) => {
  const propsErr = checkPropTypes(
    component.propTypes,
    expectedProps,
    "props",
    component.name
  );
  return propsErr;
};
