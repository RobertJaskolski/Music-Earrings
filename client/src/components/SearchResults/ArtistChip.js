import React from "react";
import PropTypes from "prop-types";
import Tooltip from "@material-ui/core/Tooltip";
import QueueIcon from "@material-ui/icons/Queue";
import RadioIcon from "@material-ui/icons/Radio";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { ChipArtist, DivArtist, IMGArtist, H2Artist } from "./style/style";
import {
  DivArtistPhone,
  IMGArtistPhone,
  H4ArtistPhone,
  SpanArtistPhone,
} from "./style/phone.style";

function ArtistChip(props) {
  const { name, images, auth } = props;
  const changeChip = useMediaQuery("(min-width:1000px)");
  if (!name) {
    return null;
  }
  return (
    <ChipArtist data-test='chipComponent' desktop={changeChip}>
      {changeChip ? (
        <DivArtist>
          <IMGArtist
            data-test='chipIMG'
            height='65px'
            width='65px'
            src={images[0]?.url || "/images/wrapper.jpg"}
            alt={name}
          />
          <H2Artist data-test='chipText'>{name}</H2Artist>

          {auth ? (
            <span>
              <Tooltip title='Listen artist radio'>
                <RadioIcon />
              </Tooltip>
              <Tooltip title='Add to quequ'>
                <QueueIcon />
              </Tooltip>
            </span>
          ) : (
            <span>
              <Tooltip title='Add to quequ'>
                <QueueIcon />
              </Tooltip>
            </span>
          )}
        </DivArtist>
      ) : (
        <DivArtistPhone>
          <IMGArtistPhone
            data-test='chipIMG'
            height='90px'
            width='90px'
            src={images[0]?.url || "/images/wrapper.jpg"}
            alt={name}
          />
          <H4ArtistPhone>
            {name}
            <br />
            {auth ? (
              <SpanArtistPhone>
                <RadioIcon />
                <QueueIcon />
              </SpanArtistPhone>
            ) : (
              <SpanArtistPhone>
                <QueueIcon />
              </SpanArtistPhone>
            )}
          </H4ArtistPhone>
        </DivArtistPhone>
      )}
    </ChipArtist>
  );
}

ArtistChip.propTypes = {
  name: PropTypes.string,
  images: PropTypes.array,
};

export default ArtistChip;
