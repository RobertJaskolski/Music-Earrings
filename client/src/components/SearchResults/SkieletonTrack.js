import React from "react";
import { StyledSkeletonTextTrack, ChipSkeltonTrack } from "./style/style";
import { Line } from "./style/phone.style";
import useMediaQuery from "@material-ui/core/useMediaQuery";

export default function SkieletonTrack() {
  const changeChip = useMediaQuery("(min-width:1000px)");
  return (
    <ChipSkeltonTrack>
      {changeChip ? (
        <StyledSkeletonTextTrack desktop={changeChip} variant='rect' />
      ) : (
        <div>
          <Line />
          <StyledSkeletonTextTrack
            desktop={changeChip}
            height={70}
            variant='rect'
          />
        </div>
      )}
    </ChipSkeltonTrack>
  );
}
