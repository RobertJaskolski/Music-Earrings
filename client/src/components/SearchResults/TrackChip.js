import React from "react";
import PropTypes from "prop-types";
import PlayArrowIcon from "@material-ui/icons/PlayArrow";
import PlaylistAddIcon from "@material-ui/icons/PlaylistAdd";
import QueueIcon from "@material-ui/icons/Queue";
import Tooltip from "@material-ui/core/Tooltip";
import useMediaQuery from "@material-ui/core/useMediaQuery";
import { ChipTrack, DivTrack, IMGTrack } from "./style/style";
import { DivTrackPhone, Line, TextTrackPhone } from "./style/phone.style";
function TrackChip(props) {
  const { track, addToFilters, auth } = props;
  const changeChip = useMediaQuery("(min-width:1000px)");
  const { name, album, external_urls } = track;
  if (!name) {
    return null;
  }
  return (
    <ChipTrack data-test='chipComponent' desktop={changeChip}>
      {changeChip ? (
        <DivTrack>
          <span>
            <PlayArrowIcon />
            <IMGTrack
              data-test='chipIMG'
              src={album?.images[0]?.url || "/images/wrapper.jpg"}
              width='40x'
              height='auto'
            />
            <div>{name}</div>
          </span>
          <span>
            {auth ? (
              <Tooltip
                style={
                  auth ? { display: "inline-block" } : { color: "#282828" }
                }
                title='Add to quequ'
              >
                <PlaylistAddIcon />
              </Tooltip>
            ) : (
              <Tooltip title='Listen on Spotify'>
                <a target='_blank' href={external_urls["spotify"] || ""}>
                  <img
                    alt='spotify Logo'
                    width='35px'
                    height='35px'
                    src='/images/spotifyLogo.svg'
                  />
                </a>
              </Tooltip>
            )}
            <Tooltip title='Add to filters for generate playlist'>
              <QueueIcon onClick={() => addToFilters(track)} />
            </Tooltip>
          </span>
        </DivTrack>
      ) : (
        <DivTrackPhone>
          <Line />
          <span>
            <IMGTrack
              data-test='chipIMG'
              src={album?.images[0]?.url || "/images/wrapper.jpg"}
              width='40x'
              height='auto'
            />
            <TextTrackPhone>{name}</TextTrackPhone>
          </span>
          <span>
            <div>
              {auth ? (
                <PlaylistAddIcon />
              ) : (
                <a target='_blank' href={external_urls["spotify"] || ""}>
                  <img
                    alt='spotify Logo'
                    width='35px'
                    height='35px'
                    src='/images/spotifyLogo.svg'
                  />
                </a>
              )}
              <QueueIcon onClick={() => addToFilters(track)} />
            </div>
            <PlayArrowIcon />
          </span>
        </DivTrackPhone>
      )}
    </ChipTrack>
  );
}

TrackChip.propTypes = {
  name: PropTypes.string,
  album: PropTypes.object,
};

export default TrackChip;
