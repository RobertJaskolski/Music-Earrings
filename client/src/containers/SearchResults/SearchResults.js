// Import outside
import React from 'react';
import { Grid } from '@material-ui/core';
import useMediaQuery from '@material-ui/core/useMediaQuery';
import CancelPresentationIcon from '@material-ui/icons/CancelPresentation';
import { useTranslation } from 'react-i18next';
// Import Components
import {
  ArtistChip,
  TrackChip,
  SkeletonArtist,
  SkieletonTrack,
} from '../../components/SearchResults';
// Import Styles
import { Section, Div, H1, DivX, DivNoResults } from './style/style';

function SearchResults({
  artists,
  tracks,
  loadingArtistsAndTracks,
  auth,
  clearArtistsAndTracks,
  searchText,
  artistSeeds,
  trackSeeds,
  addTrackToSeeds,
  addArtistToSeeds,
  setSearchText,
  handleOpenSuccessArtist,
  handleOpenFail,
  handleOpenSuccessTrack,
  changePlayingTrack,
}) {
  const { t } = useTranslation();
  let changeChip = useMediaQuery('(min-width:1000px)');
  const showSearch = searchText ? true : false;
  const addToFilters = (item) => {
    if (trackSeeds.length + artistSeeds.length < 5) {
      if (item.type === 'track') {
        addTrackToSeeds(item);
        handleOpenSuccessTrack();
      } else if (item.type === 'artist') {
        addArtistToSeeds(item);
        handleOpenSuccessArtist();
      }
    } else {
      handleOpenFail();
    }
  };
  return (
    <Section id='searchBox' active={showSearch}>
      {loadingArtistsAndTracks ? (
        <Grid container>
          <Grid item lg={6} xs={12}>
            <H1 active={showSearch}>{t('containers.SearchResults.artists')}</H1>
            <Div>
              <SkeletonArtist changeChip={changeChip} />
              <SkeletonArtist changeChip={changeChip} />
              <SkeletonArtist changeChip={changeChip} />
              <SkeletonArtist changeChip={changeChip} />
              <SkeletonArtist changeChip={changeChip} />
              <SkeletonArtist changeChip={changeChip} />
            </Div>
          </Grid>
          <Grid item lg={6} xs={12}>
            <H1 active={showSearch}>{t('containers.SearchResults.tracks')}</H1>
            <Div>
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
              <SkieletonTrack changeChip={changeChip} />
            </Div>
          </Grid>
          <Grid item lg={11} xs={5}></Grid>
          <Grid item lg={1} xs={1}>
            <DivX>
              <CancelPresentationIcon
                onClick={() => {
                  setSearchText('');
                  clearArtistsAndTracks();
                }}
              />
            </DivX>
          </Grid>
        </Grid>
      ) : (
        <Grid container>
          <Grid item lg={6} xs={12}>
            <H1 active={showSearch}>{t('containers.SearchResults.artists')}</H1>
            {artists?.length > 0 ? (
              <Div>
                {artists.map((item) => {
                  return (
                    !item.name.includes('feat') && (
                      <ArtistChip
                        addToFilters={addToFilters}
                        key={item.id}
                        artist={item}
                        auth={auth}
                        changeChip={changeChip}
                      />
                    )
                  );
                })}
              </Div>
            ) : (
              <DivNoResults active={showSearch}>
                <H1 active={showSearch}>
                  {t('containers.SearchResults.noResults')}
                </H1>
              </DivNoResults>
            )}
          </Grid>
          <Grid item lg={6} xs={12}>
            <H1 active={showSearch}>{t('containers.SearchResults.tracks')}</H1>
            {tracks?.length > 0 ? (
              <Div>
                {tracks.map((item) => {
                  return (
                    <TrackChip
                      key={item.id}
                      track={item}
                      auth={auth}
                      addToFilters={addToFilters}
                      changeChip={changeChip}
                      changePlayingTrack={changePlayingTrack}
                    />
                  );
                })}
              </Div>
            ) : (
              <DivNoResults active={showSearch}>
                <H1 active={showSearch}>
                  {t('containers.SearchResults.noResults')}
                </H1>
              </DivNoResults>
            )}
          </Grid>
          <Grid item lg={11} xs={5}></Grid>
          <Grid item lg={1} xs={1}>
            <DivX>
              <CancelPresentationIcon
                onClick={() => {
                  setSearchText('');
                  clearArtistsAndTracks();
                }}
              />
            </DivX>
          </Grid>
        </Grid>
      )}
    </Section>
  );
}

export default SearchResults;
