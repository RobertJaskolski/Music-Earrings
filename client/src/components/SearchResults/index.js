import ArtistChip from "./ArtistChip";
import TrackChip from "./TrackChip";
import SkeletonArtist from "./SkeletonArtist";
import SkieletonTrack from "./SkieletonTrack";
export { ArtistChip, TrackChip, SkeletonArtist, SkieletonTrack };
