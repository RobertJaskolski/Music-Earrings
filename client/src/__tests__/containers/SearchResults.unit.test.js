import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SearchResults from '../../containers/SearchResults/SearchResults';
import { testStore } from '../../utils/testsUtils';

const makeStore = (INIT_STATE = {}) => {
  const store = testStore(INIT_STATE);
  return store;
};

describe('Search results container', () => {
  test('should render component', () => {
    const store = makeStore();
    const wrapper = shallow(<SearchResults store={store} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
