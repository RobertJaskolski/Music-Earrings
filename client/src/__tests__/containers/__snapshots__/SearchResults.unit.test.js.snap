// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search results container should render component 1`] = `
<styled.section
  active={false}
  id="searchBox"
>
  <WithStyles(ForwardRef(Grid))
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      xs={12}
    >
      <styled.h1
        active={false}
      >
        containers.SearchResults.artists
      </styled.h1>
      <Styled(styled.div)
        active={false}
      >
        <styled.h1
          active={false}
        >
          containers.SearchResults.noResults
        </styled.h1>
      </Styled(styled.div)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={6}
      xs={12}
    >
      <styled.h1
        active={false}
      >
        containers.SearchResults.tracks
      </styled.h1>
      <Styled(styled.div)
        active={false}
      >
        <styled.h1
          active={false}
        >
          containers.SearchResults.noResults
        </styled.h1>
      </Styled(styled.div)>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={11}
      xs={5}
    />
    <WithStyles(ForwardRef(Grid))
      item={true}
      lg={1}
      xs={1}
    >
      <styled.div>
        <Memo(ForwardRef(CancelPresentationIcon))
          onClick={[Function]}
        />
      </styled.div>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</styled.section>
`;
